documentation/en/api-experimental-user-facing-v2-docs/api-v2-experimental.md

```diff
@@ -0,0 +1,1219 @@
+	latestAnchor := types.TipSetAnchors.Latest     // Anchor to latest tipset
+	safeAnchor := types.TipSetAnchors.Safe         // Anchor to safe tipset
+	finalizedAnchor := types.TipSetAnchors.Finalized // Anchor to finalized tipset
+	// keyAnchor := types.TipSetAnchors.Key(someTipSetKey) // Anchor to specific key
+
+
+	// Height 123, fallback enabled, default anchor (finalized)
+	heightSelectorDefaultAnchor := types.TipSetSelectors.Height(123, true, nil)
+	// Height 123, fallback enabled, explicitly anchored to finalized tipset
+	anchoredHeightFinalized := types.TipSetSelectors.Height(123, true, finalizedAnchor)
+	// Height 123, no fallback, anchored to latest tipset
+	anchoredHeightLatestExact := types.TipSetSelectors.Height(123, false, latestAnchor)
+	// Height 123, fallback enabled, anchored to safe tipset
+	anchoredHeightSafe := types.TipSetSelectors.Height(123, true, safeAnchor)
+	// Height 123, fallback enabled, anchored to a specific tipset key
+	// anchoredHeightKey := types.TipSetSelectors.Height(123, true, keyAnchor)
+
+	_, err = node.ChainGetTipSet(ctx, heightSelectorDefaultAnchor)
+	_, err = node.ChainGetTipSet(ctx, anchoredHeightFinalized)
+	_, err = node.ChainGetTipSet(ctx, anchoredHeightLatestExact)
+	_, err = node.ChainGetTipSet(ctx, anchoredHeightSafe)
+	// _, err = node.ChainGetTipSet(ctx, anchoredHeightKey)
+
+
+	// --- Creating Key-based Selectors ---
+	// keySelector := types.TipSetSelectors.Key(someTipSetKey)  // Select by specific TipSetKey
+	// _, err = node.ChainGetTipSet(ctx, keySelector)
+
+	// --- Using selectors with State methods ---
+	actorInfoFinalized, err := node.StateGetActor(ctx, "f01000", finalizedSelector) // Query actor at finalized state
+	if err != nil { fmt.Println("Error getting actor info:", err) } else { fmt.Println("Actor Balance (Finalized):", actorInfoFinalized.Balance) }
+
+	actorInfoSafe, err := node.StateGetActor(ctx, "f01000", safeSelector) // Query actor at safe state
+	if err != nil { fmt.Println("Error getting actor info:", err) } else { fmt.Println("Actor Balance (Safe):", actorInfoSafe.Balance) }
+
+	idAddr, err := node.StateGetID(ctx, "f01000", latestSelector) // Resolve ID at latest state
+	if err != nil { fmt.Println("Error getting ID addr:", err) } else { fmt.Println("Resolved ID Address:", idAddr) }
+}
+```
+
+These predefined selectors, anchors, and factory functions handle the internal complexity, making your Go code cleaner and less error-prone when interacting with the V2 APIs.
+
+## Consistency Across Selector Usage
+
+### Avoiding Inconsistent State Views
+
+When working with blockchain data, particularly for operations involving multiple state reads (like checking balances across accounts or calculating totals), it's crucial to maintain consistency in how you select the reference tipset. Using different selectors (`latest` vs `safe` vs `finalized`) or querying at different points in time for related data can lead to inconsistent views of the state and incorrect results.
```

Comment:
```suggestion
When working with blockchain data, particularly for operations involving multiple state reads (like checking balances across accounts or calculating totals), it's crucial to maintain consistency in how you select the reference tipset. Using selectors (`latest` vs `safe` vs `finalized`) or querying at different points in time for related data can lead to inconsistent views of the state and incorrect results.
```

Reply:
Thanks.  Fixed in https://github.com/filecoin-project/lotus/pull/13051/commits/bbd32817386a869a0c0df367e7fb75fb3ba1059b .  Resolving.